python/pyspark/pandas/utils.py

```diff
@@ -946,25 +950,20 @@ def spark_column_equals(left: Column, right: Column) -> bool:
     >>> spark_column_equals(sdf1["x"] + 1, sdf2["x"] + 1)
     False
     """
-    if isinstance(left, Column):
-        return left._jc.equals(right._jc)
-    elif isinstance(left, ConnectColumn):
+    if is_remote():
```

Comment:
I guess we need to check the class here ? and throw a proper error message?

Reply:
Yes, we should. Let me add the condition